libslide/src/utils/grammar.rs

```diff
@@ -224,33 +230,35 @@ pub fn unique_pats<'a>(expr: &'a InternedExprPat) -> HashSet<&'a InternedExprPat
     hs
 }
 
-pub fn normalize(expr: InternedExpr) -> InternedExpr {
+pub fn normalize(expr: RcExpr) -> RcExpr {
     match expr.as_ref() {
         Expr::BinaryExpr(BinaryExpr { op, lhs, rhs }) => {
             let partially_normalized = Expr::BinaryExpr(BinaryExpr {
                 op: *op,
-                lhs: normalize(*lhs),
-                rhs: normalize(*rhs),
+                lhs: normalize(lhs.clone()),
+                rhs: normalize(rhs.clone()),
             });
             let mut flattened_args =
-                get_flattened_binary_args(intern_expr!(partially_normalized, expr.span), *op);
+                get_flattened_binary_args(rc_expr!(partially_normalized, expr.span), *op);
             flattened_args.sort();
             unflatten_binary_expr(&flattened_args, *op, UnflattenStrategy::Left)
         }
-        Expr::UnaryExpr(UnaryExpr { op, rhs }) => intern_expr!(
+        Expr::UnaryExpr(UnaryExpr { op, rhs }) => rc_expr!(
             Expr::UnaryExpr(UnaryExpr {
                 op: *op,
-                rhs: normalize(*rhs),
+                rhs: normalize(rhs.clone()),
             }),
             expr.span
         ),
         Expr::Parend(expr) => {
-            let inner = normalize(*expr);
-            intern_expr!(Expr::Parend(inner), inner.span)
+            let inner = normalize(expr.clone());
+            let span = inner.span;
+            rc_expr!(Expr::Parend(inner), span)
         }
         Expr::Bracketed(expr) => {
-            let inner = normalize(*expr);
-            intern_expr!(Expr::Bracketed(inner), inner.span)
+            let inner = normalize(expr.clone());
+            let span = inner.span;
+            rc_expr!(Expr::Bracketed(inner), span)
```

Comment:
same as above

Reply:
because otherwise we have to clone inner since its used on the next line